scripts/cmake/vcpkg_configure_meson.cmake

```
@@ -54,13 +54,24 @@ function(vcpkg_internal_meson_generate_native_file _additional_binaries) #https:
     foreach(prog IN LISTS compiler)
         if(VCPKG_DETECTED_CMAKE_${prog}_COMPILER)
             string(REPLACE "CXX" "CPP" mesonprog "${prog}")
+            string(REPLACE "RC" "windres" mesonprog "${mesonprog}") # https://mesonbuild.com/Windows-module.html
             string(TOLOWER "${mesonprog}" proglower)
             string(APPEND NATIVE "${proglower} = '${VCPKG_DETECTED_CMAKE_${prog}_COMPILER}'\n")
         endif()
     endforeach()
     if(VCPKG_DETECTED_CMAKE_LINKER AND VCPKG_TARGET_IS_WINDOWS)
-        string(APPEND NATIVE "c_ld = '${VCPKG_DETECTED_CMAKE_LINKER}'\n")
-        string(APPEND NATIVE "cpp_ld = '${VCPKG_DETECTED_CMAKE_LINKER}'\n")
+        if (NOT VCPKG_DETECTED_CMAKE_C_COMPILER MATCHES "gcc|icc")
```

I don't love this way of doing this; using the path of the compiler as a check for which compiler is being used is... shakey at best. Especially when you have things like cc as your VCPKG_DETECTED_CMAKE_C_COMPILER. Instead, `VCPKG_DETECTED_CMAKE_C_COMPILER_ID` should be used (and `CMAKE_C_COMPILER_ID` should be exported from `z_vcpkg_get_cmake_vars`) (and do the same for `VCPKG_DETECTED_CXX_COMPILER`)

```suggestion
        if (NOT VCPKG_DETECTED_CMAKE_C_COMPILER_ID MATCHES "^(GNU|Intel)$")
```